### DISCUSSION TITLE
Running apollo client example in nodejs environment gives "ReferenceError: CloseEvent is not defined" error

### DISCUSSION THREAD

**Discussion by sal-limones** [enisdenjo/graphql-ws#255] [Q&A] [answered]
I'm trying to run both apollo server and client with graphql-ws subscriptions and can't figure out where to get the CloseEvent type for the nodejs environment.

**Comment by enisdenjo** [comment, marked as answer]
`CloseEvent` does not exist in the [`ws`](https://github.com/websockets/ws) library, it is exclusively a browser global class. You can, however, check if a value is _like_ a close event. Two properties can distinguish a close event, here's the check:
```js
function isLikeCloseEvent(val) {
  return 'code' in Object(val) && 'reason' in Object(val);
}
```

**Reply by sal-limones** [reply]
Thanks a lot, this method worked. Maybe it's a good idea to add this to the examples section of the docs?
